src/simplify_algebra.cpp

```diff
@@ -439,13 +498,48 @@ struct find_mul_add
         auto b_ins = r.instructions["b"];
         auto x_ins = r.instructions["x"];
         assert(x_ins != b_ins);
-
+    
         auto ax_ins = m.insert_instruction(ins, make_op("mul"), a_ins, x_ins);
         auto ab_ins = m.insert_instruction(ins, make_op("mul"), a_ins, b_ins);
         m.replace_instruction(ins, make_op("add"), ax_ins, ab_ins);
+
     }
 };
 
+
+struct find_slice_add_mul
+{
+    
+        auto matcher() const
+        {        
+            return match::name("mul")(match::either_arg(0, 1)(
+                match::name("add")(
+                    match::either_arg(0, 1)(
+                        match::name("slice")(match::arg(0)(
+                            fusable_split().bind("slc")
+                        )).bind("x"),
+                        match::any_of(conv_const_weights(), match::is_constant()).bind("b")),
+                    match::none_of(match::args(match::is_constant(), match::is_constant())),
+                    match::used_once()),
+                match::is_constant().bind("a")));
```

Comment:
Please break up this matcher using variables like above in the `find_dot_mul` matcher.